homeassistant/components/shelly/__init__.py

```diff
@@ -66,7 +66,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up Shelly from a config entry."""
     temperature_unit = "C" if hass.config.units.is_metric else "F"
 
-    ip_address = await hass.async_add_executor_job(gethostbyname, entry.data[CONF_HOST])
+    try:
+        ip_address = await hass.async_add_executor_job(
+            gethostbyname, entry.data[CONF_HOST]
+        )
+    except gaierror as err:
+        raise ConfigEntryNotReady from err
```

Comment:
IMHO, this should be handled by/in the upstream package.

Reply:
Do you mean `aioshelly` should take care of resolving an IP address from a hostname?

Reply:
Yes

Reply:
It's good idea, I'm closing this PR and I'll modify `aioshelly`.